src/profile-logic/merge-compare.js

```diff
@@ -217,7 +217,22 @@ export function mergeProfilesForDiffing(
 
     // We adjust the various times so that the 2 profiles are aligned at the
     // start and the data is consistent.
-    const startTimeAdjustment = -thread.samples.time[0];
+    let startTimeAdjustment = 0;
+    if (thread.samples.length !== 0) {
+      startTimeAdjustment = -thread.samples.time[0];
+    }
+    if (thread.markers.length !== 0) {
+      for (const startTime of thread.markers.startTime) {
+        if (startTime !== null) {
+          if (startTimeAdjustment === 0) {
+            startTimeAdjustment = startTime;
+          } else {
+            startTimeAdjustment = Math.min(startTime, startTimeAdjustment);
+          }
```

Comment:
My understanding of this code is that this sets the start of the profile at "0", so  `startTimeAdjustment` is usually a negative value. But here instead you're adding it, as a result the profiles won't be aligned at all.